.github/workflows/build-test.yml

```diff
@@ -0,0 +1,87 @@
+name: build-test
+
+on:
+  push:
+    paths:
+      - "**/Dockerfile"
+      - "**/docker-entrypoint.sh"
+      - genMatrix.js
+      - ".github/workflows/build-test.yml"
+
+  pull_request:
+    paths:
+      - "**/Dockerfile"
+      - "**/docker-entrypoint.sh"
+      - genMatrix.js
+      - ".github/workflows/build-test.yml"
+
+jobs:
+  gen-matrix:
+    name: generate-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Calculate file differences
+        uses: lots0logs/gh-action-get-changed-files@2.1.4
+        id: diff
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Generate testing matrix
+        uses: actions/github-script@v3
+        id: generator
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const script = require(`${process.env.GITHUB_WORKSPACE}/genMatrix.js`)
+            return script(
+              ${{ steps.diff.outputs.added }},
+              ${{ steps.diff.outputs.modified }},
+              ${{ steps.diff.outputs.renamed }},
+            );
+
+    outputs:
+      matrix: ${{ steps.generator.outputs.result }}
+
+  build:
+    if: ${{ fromJson(needs.gen-matrix.outputs.matrix) }}
+    needs: gen-matrix
+    name: build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.gen-matrix.outputs.matrix) }}
+
+    steps:
+      - name: Get short node version
+        uses: actions/github-script@v3
+        id: short-version
+        with:
+          result-encoding: string
+          script: return "${{ matrix.version }}".split('.')[0]
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build image
+        uses: docker/build-push-action@v2
```

Comment:
Is this also going to try and push to the DockerHub?

Reply:
By default, [push is false](https://github.com/docker/build-push-action#inputs), so it will not. The name does sound like it does, so it's initially confusing.

Reply:
add an explicit `false` here?